experimental/saiexperimentaldashha.h

```diff
@@ -0,0 +1,442 @@
+/**
+ * Copyright (c) 2024 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiexperimentaldashha.h
+ *
+ * @brief   This module defines SAI P4 extension  interface
+ */
+
+#if !defined (__SAIEXPERIMENTALDASHHA_H_)
+#define __SAIEXPERIMENTALDASHHA_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIEXPERIMENTALDASH_HA SAI - Extension specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute ID for DASH HA
+ */
+typedef enum _sai_dash_ha_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_DASH_HA_ATTR_START,
+
+    /**
+     * @brief HA Local IP for control traffic
+     *
+     * @type sai_ip_address_t
+     * @flags CREATE_AND_SET
+     * @default 0.0.0.0
+     */
+    SAI_DASH_HA_ATTR_CONTROL_LOCAL_IP = SAI_DASH_HA_ATTR_START,
+
+    /**
+     * @brief HA Peer IP for control traffic
+     *
+     * @type sai_ip_address_t
+     * @flags CREATE_AND_SET
+     * @default 0.0.0.0
+     */
+    SAI_DASH_HA_ATTR_CONTROL_PEER_IP,
+
+    /**
+     * @brief List of HA VIPs
+     *
+     * @type sai_dash_ha_vip_info_list_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_DASH_HA_ATTR_VIP_INFO_LIST,
+
+    /**
+     * @brief HA heartbeat interval time in ms
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_DASH_HA_ATTR_HEARTBEAT_INTERVAL,
+
+    /**
+     * @brief Number of HA heartbeats missed before declaring Peer unreachable
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_DASH_HA_ATTR_HEARTBEAT_COUNT,
+
+    /**
+     * @brief Time in ms required for network to converge after switchover
+     *
+     * @par Any traffic directed to wrong DPU after this time will be blackholed
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 100
+     */
+    SAI_DASH_HA_ATTR_VIP_CONVERGENCE_TIMEOUT,
```

Comment:
call this SWITCHOVER_TIMEOUT.
add a usage comment to indicate planned switchover